src/Init/Data/BitVec/Bitblast.lean

```diff
@@ -1359,6 +1359,60 @@ theorem negOverflow_eq {w : Nat} (x : BitVec w) :
     simp only [toInt_intMin, Nat.add_one_sub_one, Int.ofNat_emod, Int.neg_inj]
     rw_mod_cast [Nat.mod_eq_of_lt (by simp [Nat.pow_lt_pow_succ])]
 
+/--
+  Prove that signed division x.toInt / y.toInt only overflows when
+  x = intMin w and y = allOnes w (for 0 < w).
+
+  To show that this is the *only* case in which overflow happens, we refer to
+  overflow for negation (BitVec.sdivOverflow_eq_negOverflow_of_allOnes):
+  in fact, x.toInt/(allOnes w).toInt = - x.toInt, i.e.,
+  the overflow conditions are the same as negOverflow for x.
+
+  We can then reason about the signs of the operands. All these cases rely on respective
+  theorems specifying the bounds of signed division once the special cases are excluded:
+  · BitVec.toInt_ediv_toInt_lt_of_nonneg_of_nonneg when 0 < y.toInt and 0 < x.toInt
+  · BitVec.toInt_ediv_toInt_nonpos_of_nonpos_of_nonneg when 0 < y.toInt and x.toInt < 0
+  · BitVec.toInt_ediv_toInt_nonpos_of_nonneg_of_nonpos when y.toInt < 0 and 0 < x.toInt
+  · BitVec.toInt_ediv_toInt_lt_of_nonpos_of_lt_neg_one when y.toInt < -1 and x.toInt < 0
+
+  These BitVec theorems themselves rely on numerous Int.udiv_* theorems, that carefully
+  set the bounds of signed division for integers.
```

Comment:
```suggestion
  These BitVec theorems themselves rely on numerous Int.udiv_* theorems, that carefully
  propagate bounds for signed division for integers.
```

I would not use the word 'set', since that has an imperative flavour. I would say "propagate bounds", ala [bounds propagation](https://en.wikipedia.org/wiki/Propagation_of_uncertainty).

Reply:
accepted